pkg/machines/components/vmSnapshotsCreateModal.jsx

```diff
@@ -0,0 +1,165 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2020 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+import cockpit from "cockpit";
+import React from "react";
+import moment from "moment";
+
+import { Modal, FormGroup, HelpBlock } from "patternfly-react";
+import {
+    Button,
+    TextArea,
+    TextInput
+} from "@patternfly/react-core";
+
+import { ModalError } from "cockpit-components-inline-notification.jsx";
+import { createSnapshot } from "../libvirt-dbus.js";
+import { getVmSnapshots } from '../actions/provider-actions.js';
+
+const _ = cockpit.gettext;
+
+const NameRow = ({ onValueChanged, dialogValues, onValidate }) => {
+    return (
+        <>
+            <label className="control-label" htmlFor="name">
+                {_("Name")}
+            </label>
+            <FormGroup validationState={dialogValues.validationError.name ? "error" : undefined} controlId="name">
+                <TextInput value={dialogValues.name}
+                    id="name"
+                    type="text"
+                    onChange={(value) => onValueChanged("name", value)}
+                    onBlur={() => onValidate("name")}
```

Comment:
"I also experimented here with how we evaluate incorrect inputs in dialogs, and tried to us React's onBlur trigger. This means that whenever uses looses trigger of a input field (clicks outside of a input field or start typing into another input field), input field is evaluated. This is more immediate and more direct compared to evaluating inputs only after user click "Create"."

I chatted with Jenn a little bit about this. While the [Patternfly guidelines for error validation](https://www.patternfly.org/v4/design-guidelines/usage-and-behavior/forms#errors-and-validation) suggests both methods, without taking a strong stance on either, we tend to only use Error-on-submit in Cockpit. However, Composer uses error-on-focus, and it would be good to be consistent between the pages.
Could we skip this for now, until we get the chance to take a proper decision for all across Cockpit? I think it would be bad for the Machines page itself to be internally inconsistent.

Reply:
Alright. I was thinking about rewriting it everywhere else, so I was think about experimenting with it here.
But we can have separate PR where we do it all at once.